File: programmers/javascript/level2/n진수게임.js
const n = 16;
const t = 16;
const m = 2;
const p = 2;

const solution = (n, t, m, p) => {
  let answer = "";
  let str = "";
  let i = 0;
  while (str.length <= t * m) {
    const baseStr = i.toString(n);
    str += baseStr;
    i++;
  }
  for (let i = 0; i < str.length; i++) {
    if (answer.length === t) break;
    if ((i + 1) % m === p || ((i + 1) % m === 0 && m === p)) {
      answer += str[i];
    }
  }
  return answer.toUpperCase();
};

console.log(solution(n, t, m, p));

File: python/programmers/level2/N진수게임.py
'''
n : 진법
t : 미리 구할 숫자의 개수
m : 게임에 참가하는 인원
p : 튜브의 순서
'''
DIGITS = list('0123456789ABCDEF')


# 진법 변환 함수
def numberToBase(n, b):
    if n == 0:
        return [0]
    digits = []
    while n:
        digits.append(DIGITS[n % b])
        n = int(n // b)
    return digits[::-1]


def solution(n, t, m, p):
    answer = ''
    words = ""
    num = 0
    # 단어의 길이가 사람 수 * 필요한 글자수일때까지 실행
    while len(words) < t * m:
        words += ''.join(map(str, numberToBase(num, n)))
        num += 1
    print(words)
    for i in range(t):
        answer += words[i * m + p - 1]
    return answer


print(solution(16, 16, 2, 1))
